randovania/games/am2r/exporter/hint_namer.py

```diff
@@ -0,0 +1,124 @@
+import dataclasses
+from enum import Enum
+
+from randovania.exporter.hints.hint_formatters import LocationFormatter, TemplatedFormatter, RelativeAreaFormatter
+from randovania.exporter.hints.hint_namer import HintNamer, PickupLocation
+from randovania.exporter.hints.pickup_hint import PickupHint
+from randovania.exporter.hints.relative_item_formatter import RelativeItemFormatter
+from randovania.game_description import default_database
+from randovania.game_description.db.pickup_node import PickupNode
+from randovania.game_description.db.region_list import RegionList
+from randovania.game_description.game_patches import GamePatches
+from randovania.game_description.hint import Hint, HintLocationPrecision
+from randovania.game_description.resources.item_resource_info import ItemResourceInfo
+from randovania.games.game import RandovaniaGame
+from randovania.interface_common.players_configuration import PlayersConfiguration
+
+
+class AM2RColor(Enum):
+    WHITE = "{white}"
+    YELLOW = "{yellow}"
+    RED = "{red}"
+    PINK = "{pink}"
+    GREEN = "{green}"
+    BLUE = "{blue}"
+    LIGHT_BLUE = "{light_blue}"
+    DIM_BLUE = "{dim_blue}"
+
+
+def _area_name(region_list: RegionList, pickup_node: PickupNode, hide_region: bool) -> str:
+    area = region_list.nodes_to_area(pickup_node)
+    if hide_region:
+        return area.name
+    else:
+        return region_list.area_name(area)
+
+
+def colorize_text(color: AM2RColor, text: str, with_color: bool):
+    if with_color:
+        return f"{color.value}{text}{AM2RColor.WHITE.value}"
```

Comment:
huh, does am2r really color its text this way? that's remarkably similar to dread's system lol

Reply:
AM2R doesn't color log text at all. I'll have to write my own drawing text routine, which needs to be backwards compatible with the existing text in the game, and figured this was a good enough solution.